### PR TITLE
ModifiedUtc & PublishedUtc should take into account when recipe runs

### LINKED ISSUE #1121: ModifiedUtc and PublishedUtc are not correctly imported by a setup recipe
If I use a recipe to create content items that have `ModifiedUtc` and `PublishedUtc` properties, they don't seem to be imported correctly. 

The generated content items have a last modified date corresponding to the date of execution of the setup.

### PR DESCRIPTION
**Pull request by hishamco**
Fix #1121

@agriffard you can test it after adding `ModifiedUtc` and `PublishedUtc` properties into a recipe file, I will try to add unit test if it's required

### CODE REVIEW

**Comment by hishamco** [comment]
Even if you add some properties, not all are required

**Comment by agriffard** [comment]
> Even if you add some properties, not all are required

I know that. It is just that when you export the contents with a deployment plan and then make it a recipe, every properties are exported.

**Comment by hishamco** [comment]
Aha, coz I didn't try export yet. Anyhow I'm trying to add a unit tests if it's possible

**Comment by hishamco** [comment]
:shipit:

**Comment by hishamco** [comment]
OK, waiting for your feedback Seb

**Comment by hishamco** [comment]
@sebastienros please your feedback to close this one

**Comment by hishamco** [comment]
⌚️ 📆

File: src/OrchardCore/OrchardCore.ContentManagement/Handlers/ContentsHandler.cs [hunk 1]
```diff
@@ -64,13 +68,11 @@ public override Task PublishingAsync(PublishContentContext context)
         {
             var utcNow = _clock.UtcNow;
 
-            // The first time the content is published, reassign the CreateUtc value
             if (!context.ContentItem.PublishedUtc.HasValue)
             {
-                context.ContentItem.CreatedUtc = utcNow;
+                context.ContentItem.PublishedUtc = utcNow;
```
[the remark this answers is not in the document]
Comment (hishamco) [reply src/OrchardCore/OrchardCore.ContentManagement/Handlers/ContentsHandler.cs]: It's already set in the constructor, that's why I remove it from here

**Comment by hishamco** [comment]
> But in this case I assume you can just overwrite the values that the handlers have changes, but in the content step.

Done!! but again while we overwrite the values in `ContentStep` that indicates there's an issue with the handler that we should change in the future

**Comment by sebastienros** [comment]
The issue is that there is a single step on the recipes, "content" and it doesn't say if it's about creating a new one, or updating it. Also there is behavior by design which is about marking the CreatedUtc as the new time if the version was not published before.

Can you not change the class names, and maybe rename the file instead to prevent this weird name conflicts that make you change the namespaces?

**Comment by sebastienros** [comment]
Right now we'll just assume that if a property is "set" in the recipe, then we should just take it as is. Another option would have been to have a custom Import method in the content manager, that would not invoke the handlers.

**Comment by hishamco** [comment]
OK I will revert these changes

File: src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs [hunk 2]
```diff
@@ -43,8 +43,9 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                     await _contentManager.CreateAsync(contentItem);
                     
                     // Overwrite ModifiedUtc & PublishedUtc values that handlers have changes
-                    context.ContentItem.ModifiedUtc = contentItem.ModifiedUtc;
-                    context.ContentItem.PublishedUtc = contentItem.PublishedUtc;
+                    // Should not be necessary if IContentManager had an Import method
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs]: I just fix this by adding `Import` to `IContentManager`, please hold on, I need to write a unit test to ensure that everything work as expected

Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs]: No please. This is a much more complex task, that needs some serious design. It was removed from O1 for specific reasons.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs]: OK, it's fine

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs]: At least the CI should pass

File: src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs [hunk 3]
```diff
@@ -41,6 +41,11 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                     // Initializes the Id as it could be interpreted as an updated object when added back to YesSql
                     contentItem.Id = 0;
                     await _contentManager.CreateAsync(contentItem);
+                    
+                    // Overwrite ModifiedUtc & PublishedUtc values that handlers have changes
+                    // Should not be necessary if IContentManager had an Import method
+                    contentItem.ModifiedUtc = contentItem.ModifiedUtc;
+                    contentItem.PublishedUtc = contentItem.PublishedUtc;
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs]: @sebastienros I'm not sure how these lines will affect on the content item after it has been store in the handler?!!

**Comment by hishamco** [comment]
Now I can confirm that it's fine, @agriffard you can give it a try to confirm what I'm saying

**Comment by hishamco** [comment]
I think there's a bug here

BlogPosts

the modified date doesn't updated, while it's showing the correct date when I print it out on the front end

**Comment by hishamco** [comment]
@sebastienros is any sort of caching for that control, coz the published date is the same?!!

**Comment by hishamco** [comment]
⌚️ 📆

**Pull request merged**